src/app/modules/submission/email-submission/email-submission.controller.ts

```diff
@@ -30,267 +32,274 @@ const logger = createLoggerWithLabel(module)
+            return error
+          }),
       )
-      break
-    }
-    case AuthType.MyInfo: {
-      const uinFinResult = MyInfoUtil.extractMyInfoCookie(req.cookies)
-        .andThen(MyInfoUtil.extractAccessTokenFromCookie)
-        .andThen((accessToken) => MyInfoFactory.extractUinFin(accessToken))
-      if (uinFinResult.isErr()) {
-        const { errorMessage, statusCode } = mapRouteError(uinFinResult.error)
-        return res
-          .status(statusCode)
-          .json({ message: errorMessage, spcpSubmissionFailure: true })
-      }
-      const uinFin = uinFinResult.value
-      const verifyMyInfoResult = await MyInfoFactory.fetchMyInfoHashes(
-        uinFin,
-        formId,
-      ).andThen((hashes) =>
-        MyInfoFactory.checkMyInfoHashes(parsedResponses, hashes),
+      .andThen((form) =>
+        // Validate responses
+        EmailSubmissionService.validateAttachments(req.body.responses)
+          .andThen(() =>
+            SubmissionService.getProcessedResponses(form, req.body.responses),
+          )
+          .map((parsedResponses) => ({ parsedResponses, form }))
+          .mapErr((error) => {
+            logger.error({
+              message: 'Error processing responses',
+              meta: logMeta,
+              error,
+            })
+            return error
+          }),
       )
-      if (verifyMyInfoResult.isErr()) {
-        logger.error({
-          message: 'Error verifying MyInfo hashes',
-          meta: logMeta,
-          error: verifyMyInfoResult.error,
-        })
-        const { errorMessage, statusCode } = mapRouteError(
-          verifyMyInfoResult.error,
+      .andThen(({ parsedResponses, form }) => {
+        const { authType } = form
+        switch (authType) {
+          case AuthType.CP:
+            return SpcpFactory.extractJwt(req.cookies, authType)
+              .asyncAndThen((jwt) => SpcpFactory.extractCorppassJwtPayload(jwt))
+              .map((jwt) => ({
+                form,
+                parsedResponses: [
+                  ...parsedResponses,
+                  ...createCorppassParsedResponses(jwt.userName, jwt.userInfo),
+                ],
+                hashedFields: new Set<string>(),
+              }))
+              .mapErr((error) => {
+                spcpSubmissionFailure = true
+                logger.error({
+                  message: 'Failed to verify Corppass JWT with auth client',
+                  meta: logMeta,
+                  error,
+                })
+                return error
+              })
+          case AuthType.SP:
+            return SpcpFactory.extractJwt(req.cookies, authType)
+              .asyncAndThen((jwt) => SpcpFactory.extractSingpassJwtPayload(jwt))
+              .map((jwt) => ({
+                form,
+                parsedResponses: [
+                  ...parsedResponses,
+                  ...createSingpassParsedResponses(jwt.userName),
+                ],
+                hashedFields: new Set<string>(),
```

Comment:
since this is a refactor, should we just not pass in hashedFields if we do not need to, and let `SubmissionEmailObj` assign a default?

Reply:
agreed, this is a cleaner implementation; i did that originally for clarity to see how `hashedFields` was being used but i think it's clear regardless